app/src/components/map/components/DrawControls2.tsx

```diff
@@ -0,0 +1,185 @@
+import { useLeafletContext } from '@react-leaflet/core';
+import { Feature } from 'geojson';
+import L from 'leaflet';
+import 'leaflet-draw';
+import 'leaflet-draw/dist/leaflet.draw.css';
+import { forwardRef, useCallback, useEffect, useImperativeHandle, useRef } from 'react';
+
+/**
+ * Custom subset of `L.Control.DrawConstructorOptions` that omits `edit.featureGroup` as this will be added automatically
+ * by `DrawControls`.
+ *
+ * @export
+ * @interface IDrawControlsOptions
+ */
+export interface IDrawControlsOptions {
+  position?: L.ControlPosition;
+  draw?: L.Control.DrawOptions;
+  edit?: Omit<L.Control.EditOptions, 'featureGroup'>;
+}
+
+export interface IDrawControlsProps {
+  /**
+   * Options to control the draw/edit UI controls.
+   *
+   * @type {IDrawControlsOptions}
+   * @memberof IDrawControlsProps
+   */
+  options?: IDrawControlsOptions;
+  /**
+   * Fired each time an item is drawn (a layer is added).
+   *
+   * @memberof IDrawControlsProps
+   */
+  onLayerAdd: (event: L.DrawEvents.Created, leaflet_id: number) => void;
+  /**
+   * Fired each time an item (layer) is edited.
+   *
+   * @memberof IDrawControlsProps
+   */
+  onLayerEdit: (event: L.DrawEvents.Edited) => void;
+  /**
+   * Fired each time an item (layer) is deleted.
+   *
+   * @memberof IDrawControlsProps
+   */
+  onLayerDelete: (event: L.DrawEvents.Deleted) => void;
+}
+
+export interface IDrawControlsRef {
+  /**
+   * Adds a GeoJson feature to a new layer in the draw controls layer group.
+   *
+   * @memberof IDrawControlsRef
+   */
+  addLayer: (feature: Feature, layerId: (id: number) => void) => void;
+
+  deleteLayer: (layerId: number) => void;
+}
+
+const DrawControls2 = forwardRef<IDrawControlsRef | undefined, IDrawControlsProps>((props, ref) => {
```

Comment:
JSdoc? At least maybe an explanation on why this is needed instead of classic `DrawControl.tsx`